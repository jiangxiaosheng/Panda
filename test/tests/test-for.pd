func main() {
	var sum = 0

    for var i = 0; i < 10; i += 1 {

        if i == 5 {
            continue
        }

        sum += i

        if sum > 10 {
            break
        }
    }

}