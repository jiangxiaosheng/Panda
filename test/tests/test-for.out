Sementically checked program: 


void main() {
var sum: int(int : 0)
for (var i: int(int : 0)
; (bool : (int : i) < (int : 10)); (int : i += (int : 1))) {
if ((bool : (int : i) == (int : 5))) {
continue()
}
(int : sum += (int : i))
if ((bool : (int : sum) > (int : 10))) {
break()
}
}
}